---
title: "Analyzing CHEK1 signaling in CLL expression signatures"
author: "Junyan Lu"
date: "2021-10-26"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

# Load packages

Packages
```{r package, message=FALSE, warning=FALSE}
library(RColorBrewer)
library(DESeq2)
library(limma)
library(piano)
library(Biobase)
library(tidyverse)
source("../code/utils.R")
knitr::opts_chunk$set(dev=c("png","pdf"))
```

# BCR-triggering signature by anti-IgM stimulation for 390 min  (GSE3941)

```{r, cache=FALSE}
load("../data/RNAseq/gse39411.RData")
gmts = list(H = "../data/RNAseq/h.all.v6.2.symbols.gmt")
setMap <- read_tsv("../data/RNAseq/setToPathway.txt", col_types = "cc")
```

```{r, cache=FALSE}
#gse <- gse39411[[1]]

#subset for only B-cell and CLL without transfection
#gse <- gse[,gse$`transfected with:ch1` == "none" &
#             gse$`time point (min):ch1` == "T390" &
#             gse$`cell type:ch1` == "chronic lymphocytic leukemia B-cell"]
#save
#save(gse, file = "../data/RNAseq/gse39411.RData")


#vst
gse.vst <- gse
exprs(gse.vst) <- limma::normalizeVSN(gse.vst)

patAnno <- pData(gse.vst) %>% rownames_to_column("sampleID") %>%
  select(sampleID, description) %>%
  separate(description, into = c("patID","stimulation","timePoint"),sep = "_") %>%
  mutate(cellType = substr(patID,1,nchar(patID)-1)) %>%
  mutate(cellType = ifelse(cellType == "N", "B-cell",cellType)) %>%
  mutate(timePoint = abs(as.integer(gsub("T","",timePoint)))) %>%
  mutate(stimulation = factor(stimulation, levels = c("US","S")))

pData(gse.vst) <- patAnno[match(colnames(gse.vst), patAnno$sampleID),]
```

Differential expression (CLL cells for 390 min)
```{r}
gse.test <- gse.vst
```


Enrichment
```{r, fig.height=4, fig.width=8}
highSet <- c("UV response up", "DNA repair", "G2M checkpoint", "E2F targets")

exprMat <- exprs(gse.test)
designMat <- model.matrix( ~ 1  + patID  + stimulation, pData(gse.test))

enrichRes <- runCamera(exprMat, designMat, gmts$H, id = fData(gse.test)$`Gene Symbol`,
                       method = "camera", pCut = 0.05, ifFDR = TRUE, removePrefix = "HALLMARK_",
                       plotTitle = "anti-IgM treatment",insideLegend = TRUE, setToHighlight = highSet, setMap = setMap)
plotEnrich.IgM <- enrichRes$enrichPlot+ theme(plot.title = element_text(size=18, face = "bold"))
plotEnrich.IgM

```

# Pathway enrichment analysis using ICGC datasets

Load and pro-processing RNAseq dataset
```{r load ICGC, message=FALSE, warning=FALSE}
load("../data/RNAseq/ddsCampo.RData")
ddsCLL <- ddsCampo[,!is.na(ddsCampo$IGHV) ]
ddsCLL <- estimateSizeFactors(ddsCLL)
ddsCLL <- ddsCLL[rowSums(counts(ddsCLL)) > 10,]
ddsCLL.voom <- ddsCLL
assay(ddsCLL.voom) <- limma::voom(counts(ddsCLL), lib.size = ddsCLL$sizeFactor)$E
```

### Enrichment analysis

```{r, fig.height=10, fig.width=8}
ighv <- factor(ddsCLL.voom$IGHV, levels = c("mutated","unmutated"))
designMat <- model.matrix(~ 1 + ighv)
enrichRes <- runCamera(assay(ddsCLL.voom), designMat, gmts$H, 
                       id = rowData(ddsCLL.voom)$symbol, 
                       method = "camera", pCut = 0.01, ifFDR = TRUE, removePrefix = "HALLMARK_",
                       plotTitle = "ICGC-CLL cohort (Hallmarks, 5% FDR)", 
                       setToHighlight = highSet, setMap = setMap, insideLegend = TRUE)
enrichICGC <- enrichRes$enrichPlot
enrichICGC

```
